tests/osm/test_validate_osm.py

```diff
@@ -0,0 +1,506 @@
+                targets=dict(),
+                _list=list(),
+                search_key="unacceptable_key",
+                accepted_keys=["list", "of", "keys"],
+            )
+        # check catching conditions where search key doesn't match the name
+        # of any dictionary target
+        empty_dict = {"some_dict": dict()}
+        with pytest.raises(
+            KeyError,
+            match=re.escape(
+                "`search_key`: no_match did not match keys in `targets`:"
+                " dict_keys(['some_dict'])"
+            ),
+        ):
+            _filter_target_dict_with_list(
+                targets=empty_dict,
+                _list=[1, 2, 3],
+                search_key="no_match",
+                accepted_keys=["no_match"],
+            )
+        # this check is to simulate providing the incorrect search key
+        # resulting in an empty dictionary returned.
+        with pytest.raises(
+            ValueError,
+            match="No tags found. Did you specify the correct search_key?",
+        ):
+            _filter_target_dict_with_list(
+                targets=empty_dict,
+                _list=[1, 2, 3],
+                search_key="some_dict",
+                accepted_keys=["some_dict"],
+            )
+
+    def test__filter_target_dict_with_list_returns(self):
+        """Assert that target dict is filtered."""
+        target_dicts = {
+            "choose_me": {1: "keep", 2: "keep", 3: "keep", 4: "remove"},
+            "do_not_choose_me": {1: "should", 2: "not", 3: "appear"},
+        }
+        out = _filter_target_dict_with_list(
+            targets=target_dicts,
+            _list=[1, 2, 3],
+            search_key="choose_me",
+            accepted_keys=["choose_me", "do_not_choose_me"],
+        )
+        assert isinstance(out, dict)
+        assert "remove" not in out.values()
+
+
+def _class_atttribute_assertions(
+    some_object, some_attributes: list, some_methods: list
+) -> None:
+    """Util for checking class internals.
+
+    Asserts that the object contains the specified attributes & methods. Raises
+    AssertionError if specified attributes & methods not found within object.
+    """
+    for attr in some_attributes:
+        assert hasattr(
+            some_object, attr
+        ), f"The expected attribute `{attr}` was not found in {some_object}"
+        found_methods = [
+            getattr(some_object, m, "not_found") for m in some_methods
+        ]
+        # check all these methods were found
+        for i, method in enumerate(found_methods):
+            assert (
+                method != "not_found"
+            ), f"The expected method `{some_methods[i]}`"
+            f" was not found in {some_object}"
+        # assert they are methods
+        for i, method in enumerate(found_methods):
+            assert callable(method), f"The expected method `{some_methods[i]}`"
+            f" in {some_object} is not callable"
+
+
+class TestFindIds(object):
```

Comment:
It may be useful to add a `test_findids_raises` section where we test that the correct errors are raised when an invalid path is passed.  
This would also apply to similar classes.

Reply:
Done